csv_detective/schema_generation.py

```diff
@@ -205,31 +236,92 @@ def generate_table_schema(
         "description": "",
         "countryCode": "FR",
         "homepage": "",
-        "path": "",
-        "resources": [{"title": "", "path": ""}],
-        "sources": [],
+        "path": "https://github.com/etalab/csv-detective",
+        "resources": [],
+        "sources": [
+            {
+                "title": "Spécification Tableschema",
+                "path": "https://specs.frictionlessdata.io/table-schema"
+            },
+            {
+                "title": "schema.data.gouv.fr",
+                "path": "https://schema.data.gouv.fr"
+            }
+        ],
         "created": datetime.today().strftime("%Y-%m-%d"),
         "lastModified": datetime.today().strftime("%Y-%m-%d"),
         "version": new_version,
         "contributors": [
             {
                 "title": "Table schema bot",
-                "email": "",
-                "organisation": "Etalab",
+                "email": "schema@data.gouv.fr",
+                "organisation": "data.gouv.fr",
                 "role": "author",
             },
         ],
         "fields": fields,
         "missingValues": [""],
     }
 
-    tableschema_file = tempfile.NamedTemporaryFile(delete=False)
-    with open(tableschema_file.name, "w") as fp:
-        json.dump(schema, fp, indent=4)
+    if not save_file:
+        return schema
+
+    if save_file:
+        # Create bucket if does not exist
+        client = get_s3_client(netloc, minio_user, minio_pwd)
```

Comment:
As netloc, minio_user, minio_pwd are optional, what happens if one or more is missing?
Maybe assert betwen save_file and credentials?